pkg/scd/operational_intents_handler.go

```diff
@@ -517,6 +517,56 @@ func validateUpsertRequestAgainstPreviousOIR(
 	return nil
 }
 
+func getRelevantSubscriptionsAndIncrementIndices(
```

Comment:
Drive-by comment: also use this in `DeleteOperationalIntentReference`? There seems to be some common logic (although maybe there are some small deviations)

Reply:
Good point. I'll quickly check: if the method can be used as-is I'll update to use it, otherwise I'll keep that for another day

Reply:
Ended up doing it as it was an easy fix.